data/tutorials/ds_02_set.md

```diff
@@ -7,140 +7,253 @@ category: "data-structures"
-    val exists : (elt -> bool) -> t -> bool
-    val filter : (elt -> bool) -> t -> t
-    val filter_map : (elt -> elt option) -> t -> t
-    val partition : (elt -> bool) -> t -> t * t
-    val cardinal : t -> int
-    val elements : t -> elt list
-    val min_elt : t -> elt
-    val min_elt_opt : t -> elt option
-    val max_elt : t -> elt
-    val max_elt_opt : t -> elt option
-    val choose : t -> elt
-    val choose_opt : t -> elt option
-    val split : elt -> t -> t * bool * t
-    val find : elt -> t -> elt
-    val find_opt : elt -> t -> elt option
-    val find_first : (elt -> bool) -> t -> elt
-    val find_first_opt : (elt -> bool) -> t -> elt option
-    val find_last : (elt -> bool) -> t -> elt
-    val find_last_opt : (elt -> bool) -> t -> elt option
-    val of_list : elt list -> t
-    val to_seq_from : elt -> t -> elt Seq.t
-    val to_seq : t -> elt Seq.t
-    val to_rev_seq : t -> elt Seq.t
-    val add_seq : elt Seq.t -> t -> t
-    val of_seq : elt Seq.t -> t
-  end
-```
+# Set
+
+`Set` is a functor, which means that it is a module that is parameterized
+by another module. More concretely, this means you cannot directly create
+a set; instead, you must first specify what type of elements your set will
+contain.
 
-To create a set you need to start somewhere so here is the empty set:
+The `Set` functor provides a function `Make` which accepts a module as a
+parameter, and returns a new module representing a set whose elements have
+the type that you passed in. For example, if you want to work with sets of
+strings, you can invoke `Set.Make(String)` which will return you a new module
+which you can assign the name `SS` (short for "String Set"). Note: Be sure to
+pay attention to the case; you need to type `Set.Make(String)` and not
+`Set.Make(string)`. The reason behind this is explained in the
+"Technical Details" section at the bottom.
 
-```ocaml
-# let s = SS.empty;;
-val s : SS.t = <abstr>
+Doing this in the OCaml's toplevel will yield a lot of output:
+
+```ocamltop
+module SS = Set.Make(String);;
 ```
 
-Alternatively if we know an element to start with we can create a set
-like
+What happened here is that after assigning your newly-created module to the name
+`SS`, OCaml's toplevel displayed the module, which in this case contains
+a large number of convenience functions for working with sets (for example `is_empty`
+to check if your set is empty, `add` to add an element to your set, `remove` to
+remove an element from your set, and so on).
+
+Note also that this module defines two types: `type elt = String.t`, representing
+the type of the elements, and `type t = Set.Make(String).t`, representing the type of
+the set itself. It's important to note this because these types are used in the
+signatures of many of the functions defined in this module.
+
+For example, the `add` function has the signature `elt -> t -> t`, which means
+that it expects an element (a String), and a set of strings, and will return to you
```

Comment:
a `String.t`

Also, I would just say "will return", I find it weird to add "to you" but maybe that's just me.